packages/babel-plugin-transform-block-scoping/test/fixtures/general/issue-10046-async-generator/input.js

```diff
@@ -0,0 +1,16 @@
+// NOTE: When compiling async generators,
+// we run the async-generator-functions transform
+// as soon as in Program:enter, so when the block-scoping plugin
+// runs it has already been transpiled to a plain function.
+// The functions is thus visible to the outer scope.
+// This is a bug.
```

Comment:
Fixing this isn't easy and probably not really worth it since this a super edge case